server/monitor.go

```diff
@@ -2954,6 +2977,12 @@ func (s *Server) healthz() *HealthStatus {
 	// Check JetStream
 	js := s.getJetStream()
 	if js == nil {
+		// If JetStream should be enabled then return error status.
+		if opts.JSEnabled {
```

Comment:
This looks like it would be easy to test in one of the monitoring test that does not have JS, simply as for healthz with the option JSEnabled set to true and you should get the error.

Reply:
added a test